### DISCUSSION TITLE
Implications of inserting a Navigator

### DISCUSSION THREAD

**Discussion by knaeckeKami** [ueman/feedback#182] [Ideas]
This package inserts a Naviator near the top of the Tree - this has quite some implications.
For example, dialogs shown by `showDialog()` will now no longer be inserted under the `MaterialApp` (unless you set `useRootNavigator: false` ).

This means the dialogs will not be able to access the correct Theme and localizations (well, you can set the localization delegates in the BetterFeedback widget, so this issue has an easy workaround)

This behavior was surprising to me and it took me quite some time to figure out why my dialogs were broken.

I don't have a solution at the moment - I saw the comments in the code why a Navigator is needed. 
Maybe a NavigationKey could be passed instead of using an own navigator? 

But anyway - I think many users would appreciate it if this was documented if there is not easy way around that.

Edit: after forking and playing around with the code, I found the library works fine without a navigator - at for my use cases. 

Replacing the Navigator which 

```
  return Overlay(initialEntries: [
      OverlayEntry(builder: (context) {
          return Material( ...
```

seems to work fine with bottom sheet and TextFields. Don't know if that would break anything else though.

**Comment by ueman** [comment]
Actually this shouldn't happen anymore as the Navigator isn't inserted above the MaterialApp but rather as a neighbor so to say. Which version are you using? 

Your solution was once how I've done it, but unfortunately it didn't work out that well.

**Comment by knaeckeKami** [comment]
Using the latest published version. I'm using MateralApp.router, maybe there's a difference between just using MaterialApp. Will try to create a minimal reproducible example.
